File: Baekjoon/DP/2133.py
# https://www.acmicpc.net/problem/2133

import sys
input = sys.stdin.readline

n = int(input())
dp = [0] * (n + 1)
dp[0] = 1

for i in range(2, n + 1, 2):
    dp[i] = dp[i - 2] * 3
    for j in range(i - 4, -1, -2):
        dp[i] += dp[j] * 2
        
print(dp[n])

File: Baekjoon/DP/2133.cpp
// https://www.acmicpc.net/problem/2133

#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> dp(n + 1, 0);
    dp[0] = 1;
    dp[2] = 3;

    for (int i = 4; i <= n; i += 2) {
        dp[i] = dp[i - 2] * 3;
        for (int j = i - 4; j >= 0; j -= 2) {
            dp[i] += dp[j] * 2;
        }
    }

    cout << dp[n] << endl;

    return 0;
}